api/routes/dishesRouter.js

```diff
@@ -0,0 +1,118 @@
+const express = require('express')
+
+const dishesModel = require('../models/dishesModel.js')
+
+const router = express.Router()
+
+router.use(express.json())
+
+router.get('/', async (req, res) => {
+    try {
+        const dishes = await dishesModel.find()
+        res.status(200).json(dishes)
+    } catch (error) {
+        console.log(error)
+    }
+})
+
+router.get('/:id',  validateDishId, async (req, res) => {
+    try {
+        const dish = await dishesModel.findById(req.dish.id)
+        res.status(200).json(dish)
+    } catch (error) {
+        console.log(error)
+        res.status(500).json(error)
+    }
+})
+
+router.get('/:id/recipes',  validateDishId, async (req, res) => {
+    try {
+        const dish = await dishesModel.findById(req.dish.id)
+        const recipes = await dishesModel.findRecipesById(req.dish.id)
+        const recipesList = {
+            name: dish.name,
+            description: dish.description,
+            recipes: recipes
+        }
+        res.status(200).json(recipesList)
+    } catch (error) {
+        console.log(error)
+        res.status(500).json(error)
+    }
+})
+
+router.post('/', validateDish, async (req, res) => {
+    try {
+        const newDish = await dishesModel.add(req.validDish)
+        res.status(201).json(newDish)
+    } catch (error) {
+        console.log(error)
+        res.status(500).json(error)
+    }
+})
+
+router.put('/:id', validateDishId, validateDish, async (req, res) => {

```

Comment:
This makes sense, but it might be reasonable to _not_ validateDish here and accept a partial dish object